api/apps/articles/model.coffee

```diff
@@ -181,12 +181,13 @@ update = (article, input, callback) ->
 addSlug = (article, input, author, callback) ->
   titleSlug = _s.slugify(article.title).split('-')[0..7].join('-')
   article.slugs ?= []
+  #Don't change the article slug unless it's unpublished or a new slug is added
   if input.slug? and (input.slug != _.last(article.slugs))
     slug = input.slug
-  else if author
-    slug = _s.slugify(author.user.name) + '-' + titleSlug
+  else if article.published is false
+    slug = if author then _s.slugify(author.user.name) + '-' + titleSlug else titleSlug
   else
-    slug = titleSlug
+    return article
   article.slugs = _.unique(article.slugs).concat [slug]
   article
 
```

Comment:
Previously, when an Admin tried to click the 'Save' button in a published article after 'autosave' happened, it would generate and set a slug from the title because this method thought the second save was a duplicate from `(input.slug != _.last(article.slugs)` which caused it to generate the slug from author/title and didn't consider that it was a published article. 

Sorry, that was a mouthful.